src/shared/condition.c

```diff
@@ -138,6 +140,49 @@ static int condition_test_kernel_command_line(Condition *c) {
         return false;
 }
 
+static int condition_test_user(Condition *c) {
+        uid_t id;
+        int r;
+        _cleanup_free_ const char * username = NULL;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_USER);
+
+        r = parse_uid(c->parameter, &id);
+        if (r >= 0)
+                return id == getuid() || id == geteuid();
+        else if (r != -EINVAL)
+                return r;
+
+        username = getusername_malloc();
+        if (!username)
+                return -ENOMEM;
```

Comment:
hmm, so thinking about this: i know i suggested you to use getusername_malloc() for this, but I now think we should probably solve this differently, more matching the version of condition_test_group() you put together. Specifically: some people have multiple user entries in /etc/passwd for the same UID. (Consider: "lpoetter" and "lennart" both mapping to the same UID 1000). While setups like that are pretty broken I think in many ways, we should try to support them here as good as we can, given that it is relatively easy. Hence, we should probably try to resolve the specified user name and validate the result against the uid/euid and accept it then. i.e. if we compare resolved UIDs rather than the original strings we can nicely cover such setups.

In fact we might even want to combine both approaches, as getusername_malloc() has the benefit of checking the well-known UNIX env var $USER, and that might be worthy to keep. Hence maybe something like this:

```c
username = getusername_malloc();
if (!username)
        return -ENOMEM;
if (streq(username, c->parameter))
        return 1;

if (getpid() == 1)
        return streq(c->parameter, "root");

n = c->parameter;
r = get_user_creds(&n, &u, NULL, NULL, NULL);
if (r < 0)
        return 0;

return u == getuid() || u == geteuid();
```